aiosip/timer.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import functools
+import logging
+
+
+LOG = logging.getLogger(__name__)
+
+
+T1 = 0.5  # RTT Estimate
+T2 = 4.0  # The maximum retransmit interval for non-INVITE requests and INVITE responses
+T4 = 5.0  # Maximum duration a message will remain in the network
+
+
+class timer:
+    def __init__(self, name, default):
+        self.name = name
+        self.default = default
+        self.val = None
+
+    def __call__(self):
+        return self.val if self.val else self.default()
```

Comment:
I'm not sure if it's necessary to have `default` be a callable.

Reply:
The idea was to make changes to T1 propagate automatically thought the system. As in, if T1 is tuned to 0.3, then `timer_a` will automatically be 0.3 as well and `timer_b` will be 19.2.

I mean, I'm not married to the implementation, and we might be able to do something cleaner - I think I focused a little to closely to trying to make as many of these timers try and act like variables.

Reply:
It make sense. I don't think we will need any time soon to customise timers so implementation is up to you.